src/tests/test_definitions.py

```diff
@@ -20,17 +22,41 @@
     'noqa',
     'sections',
 ])
-def test_complex_file(test_case):
+def test_all_interpreters(test_case):
+    """Complex test cases that run under all interpreters."""
+    run_case(test_case)
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 6),
+    reason="f-string support needed"
+)
+def test_fstrings():
+    """Run the f-string test case under Python 3.6+ only."""
+    # When run under Python 3.5, mypy reports a parse error for the test file,
```

Comment:
Doesn't mypy have a file-ignore functionality? This seems overly complex. We can alternatively just avoid running mypy on Python < 3.6.

Reply:
What about this?

> A hacky way to make mypy ignore the module would be to add a stub file to the same directory, e.g. if you want mypy to ignore foo.py, add an empty foo.pyi.

https://github.com/python/mypy/issues/964

Reply:
I can try that and see if it works. If it doesn't, are you happy with this solution or should I think again?

Reply:
Thinking again could involve `eval()` of a triple-quoted string. Which ends up being a separate, parallel test harness. I thought this was nicer for using one test harness with just this filename trick.

Reply:
Not running mypy on <Python 3.6 means that mypy will not catch things like use of APIs or parameters that were added in 3.6. I assumed you would be unhappy with that.

Reply:
Since we have dropped support for py<3.6 I was able to remove all this logic.